src/os/filestore/LFNIndex.cc

```diff
@@ -449,7 +449,12 @@ int LFNIndex::list_objects(const vector<string> &to_list, int max_objs,
   }
 
   if (handle && !end) {
-    *handle = telldir(dir);
+    long cur_pos  = telldir(dir);
```

Comment:
there is only one chance telldir fail, the EBADF, seems not possible here,
i will close the PR now